customer.go

```diff
@@ -17,75 +19,31 @@ type Customer struct {
 }
 
 // Create a customer
-func CreateCustomer(client *Client, data *Customer) {
-	values := map[string]interface{}{
-		"auth":         client.Auth,
-        "api_key":      client.APIKey,
-        "firstname":    data.Firstname,
-        "lastname":     data.Lastname,
-        "email":        data.Email,
-        "phone":        data.Phone,
-        "user_id":      data.UserId,
-        "notes":        data.Notes,
-        "location_id":  data.LocationId,
-	}
-
+func CreateCustomer(data *Customer) map[string]interface{} {
     endpoint := APIBaseUrl + "customers/create"
```

Comment:
You're following a fairly standard method for REST APIs, I'd consider creating a base `Object` struct that you can then embed the particular implementations of your various data objects into. 

Also, on instantiation of each of these objects, you know that the `APIBaseUrl` will need a suffix of the data object you're manipulating, so you could set that. 

One more thing, consider using another library like (https://github.com/go-resty/resty)[resty] to drive the API, as it supports a lot of nice options, such as automatic marshaling and unmarshaling.

Reply:
This one seems interesting and I'd love for you to expound on as I'm not 100% on each detail. As for this PR, maybe outside of the scope? I'd love to visit this in a future one with some clarification?

Reply:
It would be a big refactor in the way you are currently doing it. I can go over it with you on the next meeting